File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    
    int width_max=0;
    int height_max=0;
    
    for(int i=0; i<sizes.size(); i++)
    {
        int w=max(sizes[i][0],sizes[i][1]);
        int h=min(sizes[i][0],sizes[i][1]);
        
        width_max=max(width_max,w);
        height_max=max(height_max,h);
    }
    answer=width_max*height_max;
    
    return answer;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        
        int maxWid=0;
        int maxHei=0;
        
        for(int[] size :sizes)
        {
            int wid=Math.max(size[0], size[1]);
            int hei = Math.min(size[0], size[1]);
            
            if(maxWid<wid)
                maxWid=wid;
            
            if(maxHei<hei)
                maxHei=hei;
        }
        
        
        answer= maxWid*maxHei;
        return answer;
    }
}